crates/optimism/rpc/src/eth/transaction.rs

```diff
@@ -18,6 +19,16 @@ use reth_transaction_pool::{PoolTransaction, TransactionOrigin, TransactionPool}
 
 use crate::{OpEthApi, SequencerClient};
 
+/// Bundled errors variants thrown by `TransactionCompat`.
+#[derive(Clone, Debug, Display, PartialEq, Eq)]
```

Comment:
```suggestion
#[derive(Clone, Debug, Error, PartialEq, Eq)]
```
an error should always impl the `Error` trait! use the `thiserror::Error` derive macro to do this - or are we favouring using `derive_more::Error` nowadays @Rjected @mattsse ?